v4.txt

```diff
@@ -199,7 +200,19 @@ Note: make sure you parse the response of each API correctly so that it can be u
 For example, if you want to read the content of the page, you should extract the content from the response of the reader API like `content = reader_response["data"]["content"]`.
 Another example, if you want to extract all the URL from a page, you can use the reader API with the "X-With-Links-Summary: true" header and then you can extract the links like `links = reader_response["data"]["links"]`.
 
-4. Finally, write the complete code, including input loading, calling the API functions, and saving/printing results;
+4. Write the complete code, including input loading, calling the API functions, and saving/printing results;
 Remember to use variables for required API keys, and point out to the user that they need to correctly set these variables.
 
+5. Finally, Jina AI API endpoints rate limits:
+Embedding & Reranker APIs (api.jina.ai/v1/embeddings, /rerank): 500 RPM & 1M TPM with API key; 2k RPM & 5M TPM with premium key
+Reader APIs:
+ - r.jina.ai: 20 RPM without key, 200 RPM with key, 1k RPM premium
```

Comment:
@hanxiao For simplification, should we enforce that the generated code always uses a key?
Same argument like for the zero-shot classifier. Simpler is better. The LLM already gets confused.
So I like the rate-limiting prompt, but I think we should remove the "without key" information